src/hal/machine.cpp

```diff
@@ -89,6 +89,18 @@ namespace os::detail {
     MINFO("Initializing heap\n");
     auto main_mem = memory().allocate_largest();
     MINFO("Main memory detected as %zu b\n", main_mem.size);
+    memory().deallocate(main_mem.ptr, main_mem.size);
+
+#ifndef PLATFORM_x86_solo5
```

Comment:
We could use something more generic from cmake ie PLATFORM_SUPPORTS_LIVEUPDATE ?